std/file.d

```diff
@@ -284,7 +284,6 @@ version (Posix) private void[] readImpl(const(char)[] name, const(FSChar)* namez
         ? min(statbuf.st_size + 1, maxInitialAlloc)
         : minInitialAlloc));
     void[] result = uninitializedArray!(ubyte[])(initialAlloc);
-    scope(failure) delete result;
```

Comment:
Here it's known to the engineer (i.e. you) that in case of failure no trace of `result` will be visible from the program. So it's safe to deallocate it. Since it's not really typed, you may use `GC.free(result)` within a `@trusted` lambda.